src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -1825,6 +1825,30 @@ private TypeCompareState compareTypesForEquality(CORINFO_CLASS_STRUCT_* cls1, CO
             return ObjectToHandle(merged);
         }
 
+        private bool isMoreSpecificType(CORINFO_CLASS_STRUCT_* cls1, CORINFO_CLASS_STRUCT_* cls2)
+        {
+            TypeDesc type1 = HandleToObject(cls1);
+            TypeDesc type2 = HandleToObject(cls2);
+
+            // If we have a mixture of shared and unshared types,
+            // consider the unshared type as more specific.
+            bool isType1CanonSubtype = type1.IsCanonicalSubtype(CanonicalFormKind.Any);
+            bool isType2CanonSubtype = type2.IsCanonicalSubtype(CanonicalFormKind.Any);
+            if (isType1CanonSubtype != isType2CanonSubtype)
+            {
+                // Only one of hnd1 and hnd2 is shared.
```

Comment:
Nit: The comments talk about `hnd1` and `hnd2`, but the variable names are cls1/cls2 or type1/type2.

Reply:
Ah, copypaste. Fixed.